tests/Operation/WatchFunctionalTest.php

```diff
@@ -508,6 +505,8 @@ public function testNoChangeAfterResumeBeforeInsert()
 
     public function testResumeMultipleTimesInSuccession()
     {
+        $this->skipIfIsShardedCluster('Cursor needs multiple getMore commands prohibiting rewinding it.');
```

Comment:
In contrast to `testRewindMultipleTimesWithResults()`, this test calls `next()` and asserts that the position is _not_ valid. That requires us to trust the exact nature of executing a single `getMore` on a replica set. I can see why that wouldn't be reliable with sharded clusters, since `getMore` could come back empty even if there are results.

I'm still a bit confused by this message, though. Would "getMore may return empty response before periodicNoopIntervalSecs on sharded clusters" be more accurate?

Reply:
Your analysis is correct, just like your suggestion for a skip message conveys this a lot better than my attempt.